app/Http/Controllers/DeviceController.php

```diff
@@ -173,19 +173,42 @@ private function deviceLinkMenu(Device $device)
             ];
         }
 
+        /** Use hostname as default target for all access links */
+        $hostname = trim($device->hostname);
+
         // Web
+        $https_target = $device->hostname;
+        if (Config::get('device_external_link.use_assigned_ip') && $device->overwrite_ip) {
+            $https_target = $device->overwrite_ip;
+        } elseif (Config::get('device_external_link.use_sysName') && strlen(trim($device->sysName))) {
+            $https_target = trim($device->sysName);
+        } else { // fallback to hostname
+            $https_target = $hostname;
+        }
```

Comment:
this code is repeated 6 times below

Reply:
I included a method for creatung the “url target” value (referring to that block) – at both files. That made the code amount smaller.